### PR TITLE
Make explicit any a linting error

### PR DESCRIPTION
**Pull request by southpolesteve**
Explicit `any` types is currently a warning, but this turns it into an error. We should avoid using any. Here are some good primers:

https://itnext.io/avoiding-any-in-typescript-advanced-types-and-their-usage-691b02ac345a
https://43081j.com/2019/02/typescript-avoid-using-any

In many cases where `any` is required,  `unknown` is a better type. If `any` is truly needed, we can add an eslint line exception with a comment justifying the exception.

### CODE REVIEW

File: src/Common/MongoProxyClient.ts [hunk 1]
```diff
@@ -448,12 +452,13 @@ export async function _createMongoCollectionWithARM(
   }
 
   try {
-    await new ResourceProviderClient(armEndpoint).putAsync(
+    return new ResourceProviderClient<DataModels.CreateCollectionWithRpResponse>(armEndpoint).putAsync(
```
Comment (southpolesteve) [inline src/Common/MongoProxyClient.ts:455]: @shmotlan This seems like a bug? The upstream code here expects a return value, but it was returning undefined.

Comment (shmotlan) [reply src/Common/MongoProxyClient.ts:455]: yes it seems like a bug as its not returning anything. do you still need to keep the `await` before `new ResourceProviderClient`?

Comment (southpolesteve) [reply src/Common/MongoProxyClient.ts:455]: No. You can omit `await` when returning a promise.

Comment (shmotlan) [reply src/Common/MongoProxyClient.ts:455]: ok lgtm

**Pull request merged**